guide_drafts/schema-in-depth.md

```diff
@@ -0,0 +1,218 @@
+Schema in Depth
+===============
+
+In this guide we're going to look at what exactly `infer_schema!`, `infer_table_from_schema!`, and `table!` do. For `table!` we will show a simplified version of the actual code that gets generated, and explain how each piece is relevant to you. If you've ever been confused about what exactly is getting generated, or what `use schema::posts::dsl::*` means, this is the right place to be.
```

Comment:
I'd add an intro sentence like

> If you've followed the Getting Started guide, you've already used some of Diesel's tools to work with database schemas.

to set the expectation that people should already know when to use these macros and where to look for more basic info.